app/models/data_repo_client.rb

```diff
@@ -0,0 +1,337 @@
+##
+# API client bindings for retrieving information about Terra Data Repo datasets/snapshots/schemas from their API
+##
+class DataRepoClient < Struct.new(:access_token, :api_root, :storage, :expires_at, :service_account_credentials)
+  # path to read-only service account JSON keyfile
+  SERVICE_ACCOUNT_KEY = !ENV['READ_ONLY_SERVICE_ACCOUNT_KEY'].blank? ? (ENV['NOT_DOCKERIZED'] ? ENV['READ_ONLY_SERVICE_ACCOUNT_KEY'] : File.absolute_path(ENV['READ_ONLY_SERVICE_ACCOUNT_KEY'])) : ''
```

Comment:
This code is the same for `fire_cloud_client.rb` and `papi_client.rb`, so let's put it in a superclass.